pallets/amm/src/tests.rs

```diff
@@ -366,3 +366,164 @@ fn remove_liquidity_with_more_liquidity_should_not_work() {
         );
     })
 }
+
+#[test]
+fn trade_should_work() {
+    new_test_ext().execute_with(|| {
+        use primitives::AMM as _;
+
+        let trader = AccountId::from(4_u64);
+
+        // create pool and add liquidity
+        AMM::add_liquidity(
+            Origin::signed(3.into()),
+            (DOT, XDOT),
+            (100_000_000, 100_000_000),
+            (99_999, 99_999),
+        )
+        .expect("Error initalizing AMM");
```

Comment:
Why it would throw error?

Reply:
it shouldn't! 🙂 I'll update them to use `asset_ok!` like the earlier examples.

Reply:
please do the same for other test-cases as well